### PR TITLE
Sending the received consensus messages to the correct PeerIDS

### PR DESCRIPTION
**Pull request by vgtom**
Hi David,
Please review the changes. This is only the sending part.
I will create a PR shortly for the receiving part.

### CODE REVIEW

**Comment by vgtom** [comment]
made base as hbbft

File: ethcore/sync/src/api.rs [hunk 1]
```diff
@@ -587,7 +587,28 @@ impl ChainNotify for EthSync {
 	}
 
 	fn send(&self, _message_type: ChainMessageType, _peer_id: usize) {
-		unimplemented!("TODO: Send message to _peer_id.");
+		//unimplemented!("TODO: Send message to _peer_id.");
+		self.network.with_context(WARP_SYNC_PROTOCOL_ID, |context| {
+			let peer_ids = self.network.connected_peers();
+			let mut my_peer_id=0;
+			for peer_id in peer_ids.into_iter(){
```
[the remark this answers is not in the document]
Comment (vgtom) [reply ethcore/sync/src/api.rs]: Not sure how to use Iterator::find here...i need to convert each peer_id to a node_id and compare...and then return the associated peer

[the remark this answers is not in the document]
Comment (dforsten) [reply ethcore/sync/src/api.rs]: You iterate over peer ids and you need to find a peer id.
Iterator::find() takes a lambda function which returns a bool. In that lambda you do what you do in the manually written loop, just shorter.

**Comment by vgtom** [comment]
Added the fixes as per your comments. Please review

File: ethcore/sync/src/api.rs [hunk 2]
```diff
@@ -586,8 +586,22 @@ impl ChainNotify for EthSync {
 		});
 	}
 
-	fn send(&self, _message_type: ChainMessageType, _peer_id: usize) {
-		unimplemented!("TODO: Send message to _peer_id.");
+	fn send(&self, _message_type: ChainMessageType, _peer_id: usize, node_id: Option<H512>) {
+		self.network.with_context(WARP_SYNC_PROTOCOL_ID, |context| {
+			let peer_ids = self.network.connected_peers();
+			let node_ids = peer_ids.iter().map(|&x|context.session_info(x).unwrap().id).collect::<Vec<Option<H512>>>();
+			let index = node_ids.into_iter().position(|r| r == node_id).unwrap();
```
Comment (afck) [inline ethcore/sync/src/api.rs]: So that panics if the required peer is not directly connected to us?

Comment (dforsten) [reply ethcore/sync/src/api.rs]: Yes, I encountered panics during setting up the test network already, reported as issue #173

**Pull request merged**